### PR TITLE
Bluetooth: Controller: Update AD Data PDU allocation configurability

### PR DESCRIPTION
**Pull request by cvinayak**
Keep the default AD Data length to Bluetooth Specification
defined minimum of 31 octets.

Update the calculation for configurable amount of shared
PDUs across advertising sets.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

### CODE REVIEW

File: subsys/bluetooth/controller/ll_sw/nordic/lll/lll_adv.c [hunk 1]
```diff
@@ -102,15 +102,46 @@ static inline bool isr_rx_ci_adva_check(uint8_t tx_addr, uint8_t *addr,
...
+			      BT_CTLR_ADV_SYNC_SET) * \
+			     PAYLOAD_FRAG_COUNT))
+
+/* Maximum advertising PDU buffers to allocate, which is the sum of minimum
+ * plus configured additional count in CONFIG_BT_CTLR_ADV_DATA_BUF_MAX.
+ */
+#if defined(CONFIG_BT_CTLR_ADV_EXT)
+#if defined(CONFIG_BT_CTLR_ADV_PERIODIC)
+#define PDU_MEM_COUNT_MAX ((PDU_MEM_COUNT_MIN) + \
+			   ((1 + CONFIG_BT_CTLR_ADV_DATA_BUF_MAX) * \
```
Comment (Thalley) [inline subsys/bluetooth/controller/ll_sw/nordic/lll/lll_adv.c:124]: Why do we only need 1 additional buffer for periodic advertisement? Wouldn't we need 1 buffer per advertising set?

File: subsys/bluetooth/controller/Kconfig [hunk 2]
```diff
@@ -462,7 +462,6 @@ config BT_CTLR_ADV_SET
 config BT_CTLR_ADV_DATA_LEN_MAX
 	int "Maximum Advertising Data Length"
 	range 31 1650
-	default 251
```
Comment (Thalley) [inline subsys/bluetooth/controller/Kconfig:465]: Hmm, I wonder how many, if any, samples that will stop working by defaulting to 31 instead of 251.

Comment (cvinayak) [reply subsys/bluetooth/controller/Kconfig:465]: Only the periodic advertising and direction finding connectionless tx samples enable the advertising extensions feature in the Controller, 31 octets are sufficient for these samples. @ppryga could confirm on the direction finding samples...

File: subsys/bluetooth/controller/ll_sw/nordic/lll/lll_adv.c (hunk 1 above)
Comment (cvinayak) [reply subsys/bluetooth/controller/ll_sw/nordic/lll/lll_adv.c:132]: When periodic advertising is supported, then the minimum periodic advertising instance count is 1 with a dependency of 1 extended advertising instance. `CONFIG_BT_CTLR_ADV_DATA_BUF_MAX` is 1 from Kconfing, hence 1 additional is added for a minimum periodic advertising instance. If more periodic advertising is enabled (count increased) then `CONFIG_BT_CTLR_ADV_DATA_BUF_MAX` should be increased based on the applications behavior of updates to PDU data.

If the Extended advertising data of all the sets is updated at the same time then both the elements of the double buffers of these sets would need allocation compared to application use cases where AD data is seldom updated. Hence, the required spare PDU buffers be configured using `CONFIG_BT_CTLR_ADV_DATA_BUF_MAX` by applications behavior.

File: subsys/bluetooth/controller/Kconfig (hunk 2 above)
Comment (ppryga) [reply subsys/bluetooth/controller/Kconfig:465]: DF samples and the functionality itself does not imply to extend the advertising data length. 31 octets is sufficient for the functionality.

File: subsys/bluetooth/controller/ll_sw/nordic/lll/lll_adv.c [hunk 3]
```diff
@@ -102,15 +102,64 @@ static inline bool isr_rx_ci_adva_check(uint8_t tx_addr, uint8_t *addr,
...
-#define PDU_MEM_FIFO_COUNT ((BT_CTLR_ADV_SET * PAYLOAD_FRAG_COUNT * 2) + \
-			    (CONFIG_BT_CTLR_ADV_DATA_BUF_MAX * \
+
+/* AD data and Scan Response Data need 2 PDU buffers each in the double buffer
+ * implementation. Allocate 3 PDU buffers plus CONFIG_BT_CTLR_ADV_DATA_BUF_MAX
+ * defined 1 PDU buffer as the minimum number of buffers that meet the legacy
+ * advertising needs. Add 1 each for Extended and Periodic Advertising, needed
+ * extra for double buffers for these is kept as configurable, by increasing
+ * CONFIG_BT_CTLR_ADV_DATA_BUF_MAX.
+ */
```
Comment (ppryga) [inline subsys/bluetooth/controller/ll_sw/nordic/lll/lll_adv.c:112]: Nit-pick: Mentioning here, what is default value in CONFIG_BT_CTLR_ADV_DATA_BUF_MAX, will require update if that value changes. Maybe remove that?

**Review by ppryga-nordic — APPROVED** [review]
I'm very happy that comments with explanation of unwritten things appear in the code.
Small nit-pick suggestion left, otherwise LGTM.

**Review by Thalley — APPROVED** [review]

**Pull request merged**